integrated_channels/canvas/client.py

```diff
@@ -78,22 +94,85 @@ def course_update_endpoint(canvas_base_url, course_id):
         )
 
     def create_content_metadata(self, serialized_data):
+        """
+        Creates a course in Canvas.
+        If course is not found, easy!  create it as usual
+        If course is found, it will have one of the following `workflow_state` values:
+                available: issue an update with latest field values
+                completed: this happens if a course has been concluded. Update it to change status
+                  to offer by using course[event]=offer (which makes course published in Canvas)
+                unpublished: still just update
+                deleted: take no action for now.
+        For information of Canvas workflow_states see `course[event]` at:
+        https://canvas.instructure.com/doc/api/courses.html#method.courses.update
+        """
         self._create_session()
 
-        # step 1: create the course
-        status_code, response_text = self._post(self.course_create_url, serialized_data)
-        created_course_id = json.loads(response_text)['id']
+        course_details = json.loads(serialized_data.decode('utf-8'))['course']
+        edx_course_id = course_details['integration_id']
+        located_course = CanvasUtil.find_course_by_course_id(
+            self.enterprise_configuration,
+            self.session,
+            edx_course_id
+        )
 
-        # step 2: upload image_url and any other details
-        self._update_course_details(created_course_id, serialized_data)
+        if not located_course:
+            # Course does not exist: Create the course
+            status_code, response_text = self._post(self.course_create_url, serialized_data)
+            created_course_id = json.loads(response_text)['id']
+
+            # step 2: upload image_url and any other details
+            self._update_course_details(created_course_id, course_details)
```

Comment:
Looks like we aren't even logging potential failures on this call. Can we at least add a log statement for failures here?